From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Fri, 23 Oct 2026 19:59:26 +1100
Subject: [PATCH] examples/asteroid: authorize logind set-user-linger

loginctl enable-linger returns EINVAL because logind's polkit action is not authorized for root/ceres. Allow org.freedesktop.login1.set-user-linger so the standard linger-users.service can create the marker and start user@1000.
---
 examples/asteroid/asteroid-desktop.nix | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/examples/asteroid/asteroid-desktop.nix b/examples/asteroid/asteroid-desktop.nix
index 54348ca4..1b32a327 100644
--- a/examples/asteroid/asteroid-desktop.nix
+++ b/examples/asteroid/asteroid-desktop.nix
@@ -200,6 +200,10 @@ in
     # Reference: https://github.com/AsteroidOS/meta-asteroid/blob/master/recipes-asteroid/asteroid-apps/files/30-org.freedesktop.login1.rules
     security.polkit.extraConfig = ''
       polkit.addRule(function(action, subject) {
+        if (action.id == "org.freedesktop.login1.set-user-linger" &&
+            (subject.user == "root" || subject.user == "ceres")) {
+          return polkit.Result.YES;
+        }
         if ((action.id == "org.freedesktop.login1.set-reboot-parameter" ||
              action.id == "org.freedesktop.login1.power-off-multiple-sessions" ||
              action.id == "org.freedesktop.login1.reboot-multiple-sessions") &&
